.github/workflows/docker-build.yml

```diff
@@ -11,8 +11,49 @@ env:
   EKS_NAMESPACE: court-listener
 
 jobs:
+  get-pr-labels:
+    runs-on: ubuntu-latest
+    outputs:
+      labels: ${{ steps.find_pr.outputs.labels }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Find merged PR by merge_commit_sha
+        id: find_pr
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const commitSha = context.sha;
+            const { data: prs } = await github.rest.pulls.list({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              state: "closed",
+              sort: "updated",
+              direction: "desc",
+              per_page: 1
+            });
+
+            const pr = prs.find(p => p.merge_commit_sha === commitSha);
+            if (!pr) {
+              core.setOutput("labels", "");
+              core.warning("No merged PR found for this commit.");
+              return;
+            }
+
+            const labels = pr.labels.map(l => l.name);
+            core.setOutput("labels", labels.join(","));
+            console.log(`PR #${pr.number} labels: ${labels.join(", ")}`);
+
   build:
+    needs: get-pr-labels
     runs-on: ubuntu-latest
+    if: >
+      ${{
+        !contains(needs.get-pr-labels.outputs.labels, 'skip-deploy') &&
+        !contains(needs.get-pr-labels.outputs.labels, 'skip-web-deploy') &&
+        !contains(needs.get-pr-labels.outputs.labels, 'skip-celery-deploy') &&
+        !contains(needs.get-pr-labels.outputs.labels, 'skip-cronjob-deploy')
+      }}
```

Comment:
Would this skip the image build and push if any of the skip labels is applied?

Is that correct? For instance, if the author adds the `skip-web-deploy` label, we’d still need the image to be built and pushed so that the Celery and cronjob deployments receive the changes, right?

Reply:
So if any of these labels are applied, it won't run this step, and I'm pretty sure that means the deploy won't run since it `needs` the `build` step. I'm not positive, but I think that's how it will work.

Reply:
You're totally right. Fixed.

Reply:
Great! Yeah now looks good. Set to auto-merge